### ISSUE TITLE
Restructure Readme to more of a tree like approach

### ISSUE THREAD

**Issue by munapower** [hyperledger/fabric-private-chaincode#734]
**Is your feature request related to a problem? Please describe.**

Currently the beginner is prompted with all the information all at once and this could lead to confusion. A user-friendly approach could be to add the required information in each readme in each branch. 
**Describe the solution you'd like**

Subdividing the readme into sections that are mentioned in the main but are documented in the branch that they are linked to.
**Describe alternatives you've considered**

**Additional context**

**Comment by mbrandenburger** [comment]
I suggest that we clean up the main [README.md](https://github.com/hyperledger/fabric-private-chaincode/blob/main/README.md) and create topic specific files in the [docs](https://github.com/hyperledger/fabric-private-chaincode/tree/main/docs) folder. 

The main readme should aim to clearly communicate what this project is about and how people can quickly start playing with it.

Some suggestions:
- Main `README.md` should keep the getting started section including setup Option 1). Here we just link to Option 2) in the `docs/`.
- Revisit the intro section and link to Architecture and Components in `docs/`.
- Dedicated trouble shooting file in `docs/`.
- Move SGX Support to `docs/`.
- Move reference guides to `docs/`.
- References can be moved to `docs/` and extended with other existing material.

**Comment by anshikavashistha** [comment]
May I work on this issue?
@mbrandenburger @munapower

**Comment by mbrandenburger** [comment]
Hi @anshikavashistha, yes please give it a try.

**Comment by anshikavashistha** [comment]
@mbrandenburger Agreed with you. We should only mention some introductory part in the [README.md](https://github.com/hyperledger/fabric-private-chaincode/blob/main/README.md) and rest of the content should be moved to [docs](https://github.com/hyperledger/fabric-private-chaincode/tree/main/docs) folder as you have mentioned. 

> I suggest that we clean up the main [README.md](https://github.com/hyperledger/fabric-private-chaincode/blob/main/README.md) and create topic specific files in the [docs](https://github.com/hyperledger/fabric-private-chaincode/tree/main/docs) folder.
> 
> The main readme should aim to clearly communicate what this project is about and how people can quickly start playing with it.
> 
> Some suggestions:
> 
> * Main `README.md` should keep the getting started section including setup Option 1). Here we just link to Option 2) in the `docs/`.
> * Revisit the intro section and link to Architecture and Components in `docs/`.
> * Dedicated trouble shooting file in `docs/`.
> * Move SGX Support to `docs/`.
> * Move reference guides to `docs/`.
> * References can be moved to `docs/` and extended with other existing material.

**Comment by munapower** [comment]
@anshikavashistha could you add here an outline of what we currently have in the readme?

Something like:
Hyperledger Fabric Private Chaincode
  1. Architecture and components
  1 .1. Overview
  1.2. Design
  1.3. Source Organization
2. Releases
3. Getting Started
   3.1. Clone Fabric Private Chaincode
4. Setup your development environment 
....
So that way we can see all the topics and decide if and how to reorg them.
Thanks!

**Comment by munapower** [comment]
@mbrandenburger so I finished creating the numbered list of all the topics in our main readme. 
1.	Architecture and components 
1.1.	Overview 
1.2.	Design 
1.3.	Source Organization
2.	Releases
3.	Getting Started 
3.1.	Clone Fabric Private Chaincode
4.	Setup your development environment
4.1.	Option 1: Using the Docker-based FPC Development Environment
4.1.1.	Pull docker images
4.1.2.	Manually build docker images
4.1.3.	Start the dev container
4.2.	Option 2: Setting up your system to do local development
4.2.1.	Intel SGX SDK and SSL
4.2.2.	Protocol Buffers
4.2.3.	Hyperledger Fabric
5.	Build Fabric Private Chaincode
5.1.	Intel SGX Attestation Support
5.1.1.	Register with Intel Attestation Service (IAS)
5.2.	FPC Playground for non-SGX environments
5.3.	Troubleshooting
5.3.1.	Docker
5.3.2.	Working from behind a proxy
5.3.3.	Environment settings
5.3.4.	Clang-format
5.3.5.	ERCC setup failures
5.3.6.	no Raft leader
5.3.7.	Working with the FPC dev container
5.3.8.	Development on Apple Mac (M1 or newer)
5.4.	Building Documentation
6.	Developing with Fabric Private Chaincode
6.1.	Your first private chaincode
6.2.	Developing and deploying on Azure Confidential Computing
7.	Reference Guides
7.1.	Management API
7.2.	FPC Shim
7.3.	FPC Client SDK
8.	Getting Help
9.	Contributions Welcome
10.	References
11.	Project Status
12.	Initial Committers
13.	Core Team FPC 1.0
14.	Sponsor
15.	License

**Comment by munapower** [comment]
@anshikavashistha do you still wish to contribute?

**Comment by anshikavashistha** [comment]
@munapower would be happy to work.
